File: geometry/euclidean-distance/two-dimension/2d-eucledian-distance.cpp
/*
In Eucledian distance the distance between two points is the square root 
of the sum of the squares of the differences of their coordinates
*/

#include <iostream>
#include <cmath>

double euclideanDistance(double x1, double y1, double x2, double y2) {
    return std::sqrt(std::pow(x2 - x1, 2) + std::pow(y2 - y1, 2));
}

int main() {
    double result = euclideanDistance(3.0, 4.0, 4.0, 3.0);
    std::cout << result << std::endl; // 1.4142135623730951
    return 0;
}

File: geometry/euclidean-distance/two-dimension/2d-eucledian-distance.cs
/*
In Eucledian distance the distance between two points is the square root 
of the sum of the squares of the differences of their coordinates
*/
using System;

class Program {
    static double EuclideanDistance(double x1, double y1, double x2, double y2) {
        return Math.Sqrt(Math.Pow(x2 - x1, 2) + Math.Pow(y2 - y1, 2));
    }

    static void Main() {
        double result = EuclideanDistance(3.0, 4.0, 4.0, 3.0);
        Console.WriteLine(result); // 1.4142135623730951
    }
}
